internal/cloud/realm/auth_test.go

```diff
@@ -78,16 +79,58 @@ func TestRealmAuthRefresh(t *testing.T) {
 		assert.Equal(t, nil, err)
 
 		// invalidate the session's tokens
-		session.AccessToken = ""
 		session.RefreshToken = session.AccessToken
+		session.AccessToken = ""
 
 		profile := mock.NewProfileWithSession(t, session)
 
 		client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
 		_, err = client.AuthProfile()
 		assert.Equal(t, realm.ErrInvalidSession{}, err)
 	})
-	// TODO: REALMC-7719 add test for expired credentials and test for ensuring profile cleared on invalid session
+
+	t.Run("with an expired access token", func(t *testing.T) {
+		u.SkipUnlessExpiredAccessTokenPresent(t)
+		t.Run("should use the refresh token to generate a new access token", func(t *testing.T) {
+			client := realm.NewClient(u.RealmServerURL())
+			u.SkipUnlessRealmServerRunning(t)
+			session, err := client.Authenticate(u.CloudUsername(), u.CloudAPIKey())
+			assert.Equal(t, nil, err)
+
+			session.AccessToken = u.ExpiredAccessToken() // Set the access token to an expired token
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "auth_refresh_test")
+			defer teardown()
+			profile.SetRealmBaseURL(u.RealmServerURL())
+			profile.SetSession(auth.Session{session.AccessToken, session.RefreshToken})
+
+			client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
+			_, err = client.AuthProfile()
+			assert.Nil(t, err)
+		})
+		t.Run("should error if both the access and refresh tokens are expired", func(t *testing.T) {
+			client := realm.NewClient(u.RealmServerURL())
+			u.SkipUnlessRealmServerRunning(t)
+			session, err := client.Authenticate(u.CloudUsername(), u.CloudAPIKey())
+			assert.Equal(t, nil, err)
+
+			session.AccessToken = u.ExpiredAccessToken()  // Set the access token to an expired token
+			session.RefreshToken = u.ExpiredAccessToken() // Set the refresh token to an random expired token
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "auth_refresh_test")
+			defer teardown()
+			profile.SetRealmBaseURL(u.RealmServerURL())
+			profile.SetSession(auth.Session{session.AccessToken, session.RefreshToken})
+
+			client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
+			_, err = client.AuthProfile()
+			assert.Equal(t, realm.ErrInvalidSession{}, err)
+
+			t.Log("and clear the session tokens")
+			assert.Equalf(t, "", profile.Session().AccessToken, "access token was not cleared")
+			assert.Equalf(t, "", profile.Session().RefreshToken, "refresh token was not cleared")
```

Comment:
```suggestion
			assert.Equalf(t, "", profile.Session().RefreshToken, "refresh token was not cleared: %q", profile.Session().RefreshToken)
```

Reply:
also might be worth just pulling out `profile.Session()` into its own `var` here for convenience

Reply:
Looks like we need to call profile.Session explicitly here to get the correct values, so I left as is

Reply:
I'm not sure how you're coming up with that, this code/test ran fine for me:

```go
		t.Run("should error if both the access and refresh tokens are expired", func(t *testing.T) {
			profile, teardown := mock.NewProfileFromTmpDir(t, "auth_refresh_test")
			defer teardown()
			profile.SetRealmBaseURL(u.RealmServerURL())
			// set the access and refresh token values to expired tokens
			profile.SetSession(auth.Session{u.ExpiredAccessToken(), u.ExpiredAccessToken()})

			client := realm.NewAuthClient(profile.RealmBaseURL(), profile)
			_, err := client.AuthProfile()
			assert.Equal(t, realm.ErrInvalidSession{}, err)

			session := profile.Session()

			t.Log("and clear the session tokens")
			assert.Equalf(t, "", session.AccessToken, "access token was not cleared")
			assert.Equalf(t, "", session.RefreshToken, "refresh token was not cleared")
		})
```

fwiw, especially with not printing the details, this comment no longer really matters.  i just want to make sure we're still on the same page and not missing something nefarious here

Reply:
Oh fair point, I created the `session` var above the preceding code block since I misinterpreted. I can make that change since I like it too